### DISCUSSION TITLE
Is there a performance difference between StateProvider.family and StateNotifierProvider who's state is a Map?

### DISCUSSION THREAD

**Discussion by angus-clark** [rrousselGit/riverpod#1949] [Q&A] [answered]
When managing the state of lets say a set of integers that are each tied to a specific ID, is there a performance difference between using the following providers:

`final provider1 = StateProvider.family.autoDispose<int, String>((ref, id) => 0)`

`final provider2 = StateNotifierProvider<MyNotifier, Map<String, int>>((ref) => MyNotifier())` where MyNotifier() handles various map operations.

**Comment by rrousselGit** [comment, marked as answer]
Nothing that should matter here

The family variant likely is a tiny bit less efficient. But not that much that you should avoid it
